python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -259,17 +263,49 @@ def _get_runs_to_dequeue(
             tag_concurrency_limits_counter = TagConcurrencyLimitsCounter(
                 tag_concurrency_limits, in_progress_runs
             )
-
             batch += queued_runs
             batch = self._priority_sort(batch)
 
+            if instance.run_coordinator.should_block_op_concurrency_limited_runs:
+                try:
+                    global_concurrency_limits_counter = GlobalOpConcurrencyLimitsCounter(
+                        instance,
+                        batch,
+                        in_progress_run_records,
+                        instance.run_coordinator.op_concurrency_slot_buffer,
+                    )
+                except:
+                    self._logger.error(
+                        "Failed to initialize op concurrency counter: \n%s",
+                        serializable_error_info_from_exc_info(sys.exc_info()),
+                    )
+                    # when we cannot initialize the global concurrency counter, we should fall back
+                    # to not blocking any runs based on op concurrency limits
+                    global_concurrency_limits_counter = None
```

Comment:
Neither option here seems great - potentially launching a bunch of runs vs. launching nothing at all. Do you think this is a likely scenario?

Reply:
I'm not sure... I know that we're querying info in a loop, every 1 second.  Will need to monitor this and see if we can make it even cheaper, maybe introduce a batch storage method?